kiwi/definitions/autoyast/upgrade_definition.rb

```diff
@@ -46,11 +46,22 @@
     end,
     :after_create => Proc.new do
       # Restoring old autoyast image which has to be updated.
-      mac = `xmllint --xpath  \"string(//domain/devices/interface/mac/@address)\" autoyast_description.xml`
-      system "sudo virsh undefine autoyast --remove-all-storage"
-      puts "generating autoyast image with mac address: #{mac}"
-      system "sudo virt-clone -o autoyast_sav -n autoyast --file /var/lib/libvirt/images/autoyast.qcow2 --mac #{mac}"
-      system "sudo virsh undefine autoyast_sav --remove-all-storage"
+      # FIXME: it's a little bit tricky and it deserves some refactoring/cleanup.
+      if ENV["AYTESTS_PROVIDER"] == "virtualbox"
+        vm_config = `VBoxManage showvminfo autoyast | grep "Config file" | cut -f2 -d:`.strip
+        vm_dir = File.dirname(vm_config)
+        system "VBoxManage unregistervm autoyast --delete"
+        FileUtils.mv vm_dir.sub("autoyast", "autoyast.sav"), vm_dir
+        system "VBoxManage registervm \"#{vm_config}\""
```

Comment:
Yes, you're right. But Veewee won't have access to log.info at this point (as it's an external program). Anyway, it's something we should improve in the near future (maybe defining the place where log messages should go using a environment variable).